Source/WebKit/UIProcess/Automation/BidiBrowsingContextAgent.h

```diff
@@ -61,6 +63,21 @@ class BidiBrowsingContextAgent final : public Inspector::BidiBrowsingContextBack
 
 private:
     enum class IncludeParentID: bool { No, Yes };
+    struct ScreenshotParameters {
+        String origin { "viewport"_s };
```

Comment:
I prefer that we use `enum class` and parse all supported values, and error out if we don't recognize it.

Ideally we should have an approved list of file formats and parse into an enum as well. Can you modify the protocol changes so that we create new protocol types for Origin and Format?

we already have Automation.CoordinateSystem which is pretty much what you are doing.